### PR TITLE
Rework pack usage

### PR DESCRIPTION
**Pull request by iHiD**

### CODE REVIEW

File: app/views/layouts/_head.html.haml [hunk 1]
```diff
@@ -1,29 +1,23 @@
 -# %meta{ "http-equiv": "Content-Security-Policy", content: csp_policy}
 
-%link{ rel: "preconnect", href: Exercism.config.website_assets_host, crossorigin: true }
-
```
Comment (iHiD) [inline app/views/layouts/_head.html.haml:4]: We don't need this now, as we request the asset on literally the next line.

File: app/views/layouts/_head.html.haml [hunk 2]
```diff
@@ -1,29 +1,23 @@
...
 / We always want this rendered at the start.
 = stylesheet_link_tag "website", "data-turbo-track": "reload"
 
-- if Rails.env.production?
-  %script{ async: true, src: "/usage/js/script.js", data: { domain: "exercism.org", api: "/usage/api/event" } }
-
 - if landing_page?
+  = javascript_include_tag('landing', type: :module, crossorigin: :anonymous, 'data-turbo-track': 'reload', 'data-turbo-eval': false, defer: true)
```
Comment (iHiD) [inline app/views/layouts/_head.html.haml:7]: This should be here, not in the `deferred` block.

File: app/views/layouts/_head.html.haml [hunk 3]
```diff
@@ -1,29 +1,23 @@
 -# %meta{ "http-equiv": "Content-Security-Policy", content: csp_policy}
 
-%link{ rel: "preconnect", href: Exercism.config.website_assets_host, crossorigin: true }
-
 / We always want this rendered at the start.
 = stylesheet_link_tag "website", "data-turbo-track": "reload"
 
-- if Rails.env.production?
-  %script{ async: true, src: "/usage/js/script.js", data: { domain: "exercism.org", api: "/usage/api/event" } }
-
```
Comment (iHiD) [inline app/views/layouts/_head.html.haml:10]: This should be below the other JS loads

File: app/views/layouts/_head.html.haml [hunk 4]
```diff
@@ -1,29 +1,23 @@
...
     %link{ href: asset_path("core.js"), rel: "prefetch", as: :script }
+    %link{ href: asset_path("application.js"), rel: "prefetch", as: :script }
 
 - else
   = stylesheet_link_tag "internal", "data-turbo-track": "reload"
   = stylesheet_link_tag "application", "data-turbo-track": "reload"
   = javascript_include_tag('core', type: :module, crossorigin: :anonymous, 'data-turbo-track': 'reload', 'data-turbo-eval': false)
   - js_packs.each do |pack|
-    = javascript_include_tag(pack, type: :module, crossorigin: :anonymous, 'data-turbo-track': 'reload', 'data-turbo-eval': false)
+    = javascript_include_tag(pack, type: :module, crossorigin: :anonymous, 'data-turbo-track': 'reload', 'data-turbo-eval': false, defer: true)
```
Comment (iHiD) [inline app/views/layouts/_head.html.haml:19]: I don't see a reason why other files can't be defered until the page is loaded as long as `core` runs immediately.

**Review by ErikSchierboom — APPROVED** [review]
LGTM!

**Pull request merged**